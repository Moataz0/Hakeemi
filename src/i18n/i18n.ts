import i18next from 'i18next';
import en from './en';
import ar from './ar';
import { format } from 'date-fns';
import { enGB as English, ar as Arabic } from 'date-fns/locale';

export const resources = {
  en,
  ar,
} as const;

const getDateLocale = (lng?: string) => {
  switch (lng) {
    case 'ar':
      return Arabic;
    case 'en':
    default:
      return English;
  }
};

const i18n = i18next.createInstance();
i18n.init({
  fallbackLng: 'en',
  lng: 'en',
  supportedLngs: ['en', 'ar'],
  defaultNS: 'common',
  resources,
  react: {
    useSuspense: false,
  },
  interpolation: {
    format: (value, pattern, lng) => {
      if (value instanceof Date && pattern) {
        return format(value, pattern, { locale: getDateLocale(lng) });
      }
      return value;
    },
  },
});

declare module 'react-i18next' {
  type DefaultResources = typeof resources['en'];
  interface Resources extends DefaultResources {}
}

export default i18n;
