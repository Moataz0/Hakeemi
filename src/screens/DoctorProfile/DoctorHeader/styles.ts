import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 30,
  },
  textContainer: { marginLeft: 10 },
});
