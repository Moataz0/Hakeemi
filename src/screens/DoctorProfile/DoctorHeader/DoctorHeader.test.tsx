import React, { FC } from 'react';
import { render } from '@testing-library/react-native';
import DoctorHeader from './DoctorHeader';
import { I18nextProvider } from 'react-i18next';
import i18n from '../../../i18n';

const wrapper: FC = ({ children }) => {
  return <I18nextProvider i18n={i18n}>{children}</I18nextProvider>;
};
describe('Button component test', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <DoctorHeader
        title="Title"
        subtitle="Subtitle"
        imagURL="http://www.test.com/image.jpg"
      />,
      { wrapper },
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
