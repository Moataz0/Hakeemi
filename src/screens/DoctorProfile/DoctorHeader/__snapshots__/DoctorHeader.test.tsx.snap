// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingHorizontal": 30,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(214, 250, 255, 1)",
          "borderRadius": 20,
          "height": 80,
          "overflow": "hidden",
          "width": 80,
        },
        undefined,
        undefined,
      ]
    }
    testID="avatar"
  >
    <Image
      source={
        Object {
          "height": 80,
          "uri": "http://www.test.com/image.jpg",
          "width": 80,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "marginLeft": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(0, 0, 0, 1)",
          "fontFamily": "System",
          "fontSize": 30,
          "fontWeight": "400",
          "includeFontPadding": false,
          "lineHeight": 38,
          "textAlign": undefined,
        }
      }
      testID="Text"
    >
      Title
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(144, 144, 144, 1)",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "includeFontPadding": false,
          "lineHeight": 18,
          "textAlign": undefined,
        }
      }
      testID="Text"
    >
      Subtitle
    </Text>
  </View>
</View>
`;
